### PR TITLE
gossipsub fixes

### PR DESCRIPTION
**Pull request by arnetheduck**
* graft up to D peers
* fix logging so it's clear who is grafting/pruning who
* clear fanout when grafting

### CODE REVIEW

File: libp2p/protocols/pubsub/gossipsub.nim [hunk 1]
```diff
@@ -99,49 +99,48 @@ proc replenishFanout(g: GossipSub, topic: string) =
...
+    grafts = toSeq(
       g.gossipsub.getOrDefault(topic, initHashSet[PubSubPeer]()) -
       g.mesh.getOrDefault(topic, initHashSet[PubSubPeer]())
     )
 
     logScope:
-      topic = topic
       meshPeers = g.mesh.peers(topic)
-      newPeers = newPeers.len
+      grafts = grafts.len
```
Comment (dryajov) [inline libp2p/protocols/pubsub/gossipsub.nim:122]: what is this doing?

Comment (arnetheduck) [reply libp2p/protocols/pubsub/gossipsub.nim:122]: printing number of graft candidates I guess

Comment (dryajov) [reply libp2p/protocols/pubsub/gossipsub.nim:122]: I think this is resetting the seq's size, but it's setting it to itself, so it's a noop, look fishy. I know it was there before you got to it, but seems wrong either way.

Comment (dryajov) [reply libp2p/protocols/pubsub/gossipsub.nim:122]: oh, it's a logscope, nevermind.

**Pull request merged**